src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/TransmissionStorage.cs

```diff
@@ -139,24 +140,32 @@ public virtual Transmission Dequeue()
                         return transmission;
                     }
                 }
-                catch (UnauthorizedAccessException)
+                catch (UnauthorizedAccessException uae)
                 {
+                    TelemetryChannelEventSource.Log.TransmissionStorageDequeueUnauthorizedAccessException(file?.Name ?? string.Empty, uae.ToString());
                     if (file == null)
                     {
-                        return null; // because the process does not have permission to access the folder
+                        return null; // Because the process does not have permission to access the folder.
                     }
 
                     if (lastInaccessibleFileName != file.Name)
                     {
                         lastInaccessibleFileName = file.Name;
-                        continue; // because another thread is loading this file right now
+                        continue; // Because another thread is loading this file right now.
+                    }
+                    else
+                    {
+                        // The same file has been inaccessible more than once.
+                        TelemetryChannelEventSource.Log.TransmissionStorageInaccessibleFile(file.Name);
                     }
 
-                    throw; // because the process does not have permission to modify this file
+                    throw; // Because the process does not have permission to modify this file.
                 }
-                catch (IOException)
+                catch (IOException ioe)
                 {
-                    continue; // because another thread already loaded this file
+                    TelemetryChannelEventSource.Log.TransmissionFailedToStoreWarning(file.Name, ioe.ToString());
+                    Thread.Sleep(random.Next(1, 100)); // Sleep for random time of 1 to 100 milliseconds to try to avoid future timing conflicts.
```

Comment:
P.S. The caller does not wait, it runs in a tight loop until there are no more files or no room in the buffer to load them.  The existing logic in the system does not allow it to really stop trying until there are no files physically in the directory.  If we can learn with the additional logic what is happening at the file system level when it fails and gets stuck in a loop then it will give us enough information to try to fix it permanently.